File: 15-Algorithms_dynamic_programming/memoization.rb
# frozen_string_literal: true

def add_to80(num)
  num + 80
end

@cache = {}

def memoized(num)
  return @cache[num] if @cache.key?(num)

  puts 'long time'
  @cache[num] = num + 80
end

puts memoized(6)
puts memoized(6)

File: 15-Algorithms_dynamic_programming/memoization.js
const addTo80 = n => n + 80;

let cache = {};

const memoized = n => {
  if (n in cache) return cache[n];

  console.log('long time');
  return (cache[n] = n + 80);
};

console.log(1, memoized(6));
console.log(2, memoized(6));
